src/cart/crt_hg_proc.c

```diff
@@ -317,16 +285,21 @@ crt_proc_corpc_hdr(crt_proc_t proc, struct crt_corpc_hdr *hdr)
 static inline int
 crt_proc_common_hdr(crt_proc_t proc, struct crt_common_hdr *hdr)
 {
-	int rc;
+	crt_proc_op_t	proc_op;
+	int		rc;
+
+	if (unlikely(hdr == NULL))
+		D_GOTO(out, rc = -DER_INVAL);
 
 	/*
 	 * D_DEBUG("in crt_proc_common_hdr, opc: %#x.\n", hdr->cch_opc);
 	 */
 
-	if (unlikely(proc == CRT_PROC_NULL || hdr == NULL))
-		D_GOTO(out, rc = -DER_INVAL);
+	rc = crt_proc_get_op(proc, &proc_op);
```

Comment:
With functions like crt_proc_get_op() which can essentially only fail if there is local memory corruption I wonder if we should simply make them assert on failure rather than return an error code.

Reply:
Maybe it make sense. Need to check about compiler behave.